src/pipelinedefinitionstatus.hpp

```diff
@@ -304,6 +304,11 @@ class PipelineDefinitionStatus : public MachineState<BeginState, AvailableState,
                (state == PipelineDefinitionStateCode::LOADING_PRECONDITION_FAILED_REQUIRED_REVALIDATION) ||
                (state == PipelineDefinitionStateCode::BEGIN);
     }
+    bool requireRevalidation() const {
+        auto state = getStateCode();
```

Comment:
This creates local copy of state code. Would it be better to call getStateCode twice below?

Reply:
state is an enum. getStateCode calls underlying std::visit which then according to its current state calls getStateCode of underlying state which returns PipelineDefinitionStateCode.